plugins/catalog-react/src/components/EntityOwnerPicker/EntityOwnerPicker.tsx

```diff
@@ -173,18 +177,25 @@ export const EntityOwnerPicker = (props?: EntityOwnerPickerProps) => {
                 }
                 onClick={event => event.preventDefault()}
                 label={
-                  <Box display="flex" flexWrap="wrap" alignItems="center">
-                    {isGroup ? (
-                      <GroupIcon fontSize="small" />
-                    ) : (
-                      <PersonIcon fontSize="small" />
-                    )}
-                    &nbsp;
-                    {humanizeEntity(
-                      entity,
-                      humanizeEntityRef(entity, { defaultKind: entity.kind }),
-                    )}
-                  </Box>
+                  <Tooltip title={humanizedName}>
+                    <Box display="flex" flexWrap="wrap" alignItems="center">
+                      {isGroup ? (
+                        <GroupIcon fontSize="small" />
+                      ) : (
+                        <PersonIcon fontSize="small" />
+                      )}
+                      &nbsp;
+                      <Box
+                        style={{
+                          overflow: 'hidden',
+                          textOverflow: 'ellipsis',
+                          width: '11rem',
```

Comment:
`width: 11rem` seems like a bit of a magic number

Reply:
It honestly is. I'll try to change it to be more dynamic